pkg/controllers/vdb/init_db.go

```diff
@@ -269,12 +274,58 @@ func (g *GenericDatabaseInitializer) setKerberosAuthParms() {
 	g.ConfigurationParams.Set("KerberosEnableKeytabPermissionCheck", "0")
 }
 
+func (g *GenericDatabaseInitializer) setAuthFromGCSSecret(ctx context.Context) (string, ctrl.Result, error) {
+	name := "projects/" + g.Vdb.Annotations[meta.GcpProjectIDAnnotation] +
+		"/secrets/" + g.Vdb.Annotations[meta.GcpSecNameAnnontation] +
+		"/versions/" + g.Vdb.Annotations[meta.GcpSecVersionAnnotation]
```

Comment:
We shouldn't use annotations to come up with then secret name. We already have a field in the VerticaDB CR for the name and should use that: `spec.communal.credentialSecret`

Reply:
Removed the new annotations and used the existing credentialSecret for GSM as well